test/ava-files.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+var test = require('tap').test;
+var AvaFiles = require('../lib/ava-files');
+
+test('testMatcher', function (t) {
+	var avaFiles = new AvaFiles(['**/foo*']);
+
+	var matcher = avaFiles.makeTestMatcher();
+
+	function isTest(file) {
+		t.true(matcher(file), file + ' should be a test');
+	}
+
+	function notTest(file) {
+		t.false(matcher(file), file + ' should not be a test');
+	}
+
+	isTest('foo-bar.js');
+	isTest('foo.js');
+	isTest('foo/blah.js');
+	isTest('bar/foo.js');
+	notTest('bar.js');
+	notTest('bar/bar.js');
+	notTest('_foo-bar.js');
+	notTest('foo/_foo-bar.js');
+	notTest('foo-bar.txt');
+	notTest('node_modules/foo.js');
+	notTest('fixtures/foo.js');
+	notTest('helpers/foo.js');
+	t.end();
+});
+
+test('sourceMatcher', function (t) {
+	var avaFiles = new AvaFiles(['**/foo*']);
```

Comment:
Yea I went commit by commit. It's clear in the full diff that this tests the defaults.